File: dp/2264-MinimumSumOfFourDigitNumberAfterSplittingDigits/2264-MinimumSumOfFourDigitNumberAfterSplittingDigits.java
// Last updated: 6/9/2026, 9:13:06 PM
class Solution {
    public int minimumSum(int num) {
        int cnt[]=new int[4];
        int i=0;
        while(i<4){
            cnt[i]=num%10;
            num/=10;
            i++;
        }
        Arrays.sort(cnt);
        int ans1=cnt[2]+cnt[3];
        int c1=ans1/10;
        int ans2=cnt[0]+cnt[1];
        int ans=(ans2+c1)*10+ans1%10;
        return ans;
    }
}

File: dp/2264-MinimumSumOfFourDigitNumberAfterSplittingDigits/2264-MinimumSumOfFourDigitNumberAfterSplittingDigits.cpp
// Last updated: 6/9/2026, 9:13:13 PM
class Solution {
public:
    int minimumSum(int num) {
        vector<int> v;
        while(num>0){
            v.push_back(num%10);
            num/=10;
        }
        sort(v.begin(),v.end());
        // for(auto i:v) cout<<i<<" ";
        string n1,n2;
        int c=0;
        for(auto i:v){
            if(c%2) n2+=(i+'0');
            else n1+=(i+'0');
            c++;
        }
        // cout<<n1<<" "<<n2<<endl;
        // cout<<<<endl;
        int ans=stoi(n1)+stoi(n2);
        return ans;
        
    }
};